overlord/fdestate/fdestate.go

```diff
@@ -204,26 +204,32 @@ func initializeState(st *state.State) error {
 	devpData := fmt.Sprintf("/dev/disk/by-uuid/%s", dataUUID)
 	devpSave := fmt.Sprintf("/dev/disk/by-uuid/%s", saveUUID)
 	digest, err := getPrimaryKeyDigest(devpData)
-	if err != nil {
-		return fmt.Errorf("cannot obtain primary key digest for data device %s: %w", devpData, err)
-	}
-	// TODO: restore key verification once we know that it is always added to
-	// the keyring
-	sameDigest, err := digest.verifyPrimaryKeyDigest(devpSave)
 	if err != nil {
 		if !errors.Is(err, secboot.ErrKernelKeyNotFound) {
-			return fmt.Errorf("cannot verify primary key digest for save device %s: %w", devpSave, err)
-		} else {
+			return fmt.Errorf("cannot obtain primary key digest for data device %s: %w", devpData, err)
+		}
+		// Some very old kernels with FDE do not set primary key in the keyring
+		logger.Noticef("cannot obtain primary key digest for data device %s: %v", devpData, err)
+		s.PrimaryKeys = map[int]PrimaryKeyInfo{}
```

Comment:
a doc comment for PrimaryKeys should probbly mention that i can be empty in some backward compat scenarios

Reply:
I have added the cases I know in the documentation of the field.